portal/views.py

```diff
@@ -77,13 +79,27 @@ def get_public_apps(request, app_id=0, get_all=True, collection=None):
             published_apps_qs = subclass.objects.filter(~Q(app_status__status="Deleted"), access="public")
             published_apps.extend([app for app in published_apps_qs])
 
-    # sorting the apps by date updated
-    published_apps.sort(
-        key=lambda app: (app.updated_on is None, app.updated_on if app.updated_on is not None else ""),
-        reverse=True,  # Sort in descending order
-    )
+    # sort the apps and prepare for the homepage view if specified
+    if order == "created_on_for_homepage":
+        published_apps = [app for app in published_apps if app.updated_on <= (app.created_on + timedelta(minutes=60))]
+        # we don't count updates in the first hour as updates, these should still be displayed under recently added apps
+        published_apps.sort(
+            key=lambda app: (app.created_on is None, app.created_on if app.created_on is not None else ""),
+            reverse=True,  # Sort in descending order
+        )
+    elif order == "updated_on_for_homepage":
+        published_apps = [app for app in published_apps if app.updated_on > (app.created_on + timedelta(minutes=60))]
+        published_apps.sort(
+            key=lambda app: (app.updated_on is None, app.updated_on if app.updated_on is not None else ""),
+            reverse=True,  # Sort in descending order
+        )
+    else:
+        published_apps.sort(
+            key=lambda app: (app.updated_on is None, app.updated_on if app.updated_on is not None else ""),
+            reverse=True,  # Sort in descending order
+        )
 
-    if len(published_apps) >= 3 and not get_all:
+    if order == "created_on_for_homepage" or order == "updated_on_for_homepage" and len(published_apps) >= 3:
```

Comment:
I have a feeling that it would be better to generalize it a little.

Like pass a `reverse` kwarg and have a time delta be an option in the function arguments

Reply:
Done.